app/controllers/staff/activities_controller.rb

```diff
@@ -39,7 +41,25 @@ def id
   end
 
   def organisation_id
-    params[:organisation_id]
+    organisation_id = params.fetch(:organisation_id, current_user.organisation)
+
+    return current_user.organisation.id unless known_organisation_id?(organisation_id)
+    return current_user.organisation.id if organisation_id.blank?
```

Comment:
I don't think this line is needed, because the `params.fetch`  is doing the same thing.

Having said that, if we're setting the `organisation_id` parameter to one value, but the user doesn't have access to that information, shouldn't we be returning a 404 *or* redirect them to the page for their organisation?

Reply:
I thought I had checked that, with the `fetch` I am trying to catch the case when there is no param, so `/activities` and defaulting to the current user org. The `blank?` is a catch for when the param is present, but there is no value so `/acticities?organisation_id` - I thought a lot about all these case and if we should try to handle them, which leads nicely to…

I considered both when we might 404, 403, redirect, etc and ended up discussing with @tahb as I couldn't make the decision on which was best. 

I feel like the 'correct' thing to do it return the error page, but from a users point of view I am not sure that is helpful, we felt missing or changing the query param is not something a user would do intentionally and so returning an error may not be the most helpful and given a delivery partner can only see their own organisations activities, showing that is a safe bet.

Let me know what you think?

Reply:
I'd argue that we should be always trying to do the correct thing - if the user isn't going to fiddle with the query parameter, then surely there's no harm in showing an error message when they do? But I'm happy for this to go in because it sounds like you've thought about all of these situations! :shipit: